Sources/QueryEngine/QueryEngine.swift

```diff
@@ -34,13 +35,13 @@ public func withQueryEngine(
 
 /// Cacheable computations engine. Currently the engine makes an assumption that computations produce same results for
 /// the same query values and write results to a single file path.
-public actor QueryEngine {
+package actor QueryEngine {
     private(set) var cacheHits = 0
     private(set) var cacheMisses = 0
 
-    public let fileSystem: any AsyncFileSystem
-    public let httpClient = HTTPClient()
-    public let observabilityScope: ObservabilityScope
+    package let fileSystem: any AsyncFileSystem
+    package let httpClient = HTTPClient()
+    package let observabilityScope: ObservabilityScope
     private let resultsCache: SQLiteBackedCache<FileCacheRecord>
     private var isShutDown = false
```

Comment:
Need to handle circular query dependencies here, probably with a dictionary from `Query` to query result continuation.

Reply:
Don't recompute `Query` hash for deep FS trees.